### PR TITLE
Setting financial goals - group. Setting financial goals - individual.

### PR DESCRIPTION
**Pull request by Svviat**
## JIRA

* https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=896&view=detail&selectedIssue=LVPYTTTN-23

## Code reviewers

- [ ] @antooa
- [ ] @HalynaDom
- [ ] @OlexaNdrus
- [ ] @pinkfoxxx
- [ ] @Rocckk
- [ ] @Rostyk1404
- [ ] @Sent1nelDNoir
- [ ] @ValentynLiashenko

### Second Level Review

- [ ] @Blackleaves
- [ ] @yevgenbehen
- [ ] @lhalam

## Summary of issue

Setting financial goals - group, setting financial goals- individual

### CODE REVIEW

File: iBudget/static/src/components/Financial_goal.vue [hunk 1]
```diff
@@ -0,0 +1,139 @@
+<template>
```
Comment (Blackleaves) [inline iBudget/static/src/components/Financial_goal.vue]: For the whole file: Indentation.

You have such indentation width here: 5, 4, 3, 2, 1 spaces in different places.
Opening and closing tags not always starts/ends on the same line.

File: iBudget/utils/validators.py [hunk 2]
```diff
@@ -118,6 +119,31 @@ def input_spending_registration_validate(data):
...
+    except (ValidationError, AttributeError):
+        return False
+
+def date_range_validate(data):
+    """
+    Function that provides data range validation
+    :param data: start_date, finish_date
+    :return: Raise validation error when the starting date is greater than the final one
+    """
+    if data["start_date"] > data["finish_date"]:
```
Comment (Blackleaves) [inline iBudget/utils/validators.py]: Could user press Save button on the page when start date is after finish date?

Comment (Svviat) [reply iBudget/utils/validators.py]: Yes, but then the validator will work and it will give an error.

File: iBudget/fund/models.py [hunk 3]
```diff
@@ -68,3 +68,23 @@ class FinancialGoal(models.Model):
...
+            value: financial aim.
+            fund: Fund for individual purpose.
+            start_date: The beginning of time period.
+            finish_date: The end of time period.
+        Returns:
+            SpendingLimitationIndividual object if row with described data exists, None otherwise.
+
+
+        """
+        goal = FinancialGoal.objects.filter(
```
Comment (yevgenbehen) [inline iBudget/fund/models.py]: Filter can return several goals so this should be called `goals`

File: iBudget/fund/views.py [hunk 4]
```diff
@@ -21,3 +29,60 @@ def show_fund(request):
             user_funds.append({'id': entry.id, 'name': entry.name})
         return JsonResponse(user_funds, status=200, safe=False)
     return JsonResponse({}, status=400)
+
+
+@require_http_methods(["GET"])
+def show_fund_group(request):
```
Comment (yevgenbehen) [inline iBudget/fund/views.py]: You have called method `show_fund_group` but it returns a list of `users_fund`. This is confusing.

File: iBudget/utils/validators.py (hunk 2 above)
Comment (Blackleaves) [reply iBudget/utils/validators.py]: For the future: maybe it will be good to implement validation on front-end too.

Comment (Svviat) [reply iBudget/utils/validators.py]: he's already there

**Pull request merged**